daemon/entrypoint.sh

```diff
@@ -415,6 +418,29 @@ elif [[ "$CEPH_DAEMON" = "OSD_CEPH_DISK" ]]; then
   exec /usr/bin/ceph-osd -f -d -i ${OSD_ID}
 
 
+##########################
+# OSD_CEPH_DISK_ACTIVATE #
+##########################
+
+elif [[ "$CEPH_DAEMON" = "OSD_CEPH_DISK_ACTIVATE" ]]; then
+
+  ceph_config_check
+
+  if [[ -z "${OSD_DEVICE}" ]];then
+    echo "ERROR- You must provide a device to build your OSD ie: /dev/sdb"
+    exit 1
+  fi
+
+  mkdir -p /var/lib/ceph/osd
+
+  ceph-disk -v activate ${OSD_DEVICE}1
+  OSD_ID=$(cat /var/lib/ceph/osd/$(ls -ltr /var/lib/ceph/osd/ | tail -n1 | awk -v pattern="$CLUSTER" '$0 ~ pattern {print $9}')/whoami)
+  OSD_WEIGHT=$(df -P -k /var/lib/ceph/osd/${CLUSTER}-$OSD_ID/ | tail -1 | awk '{ d= $2/1073741824 ; r = sprintf("%.2f", d); print r }')
```

Comment:
Actually this is the formula used in the ceph init script to set the weight of an OSD, see here: https://github.com/ceph/ceph/blob/697a20a0c542576a0e4ecf0e06414affeb64f1e0/src/init-ceph.in#L368

Basically 1073741824 bytes = 1GB.